### ISSUE TITLE
Power8: Host stuck during booting with latest devel branch(4.16.0-1.rc7.dev.git58079f0.el7)

### ISSUE THREAD

**Issue by sathnaga** [open-power-host-os/linux#30]
<cde:info> Mirrored with LTC bug https://bugzilla.linux.ibm.com/show_bug.cgi?id=166290 </cde:info>

Boot power8 with latest devel branch kernel 4.16.0-1.rc7.dev.git58079f0.el7

```
[   39.095205] systemd[1]: Detected architecture ppc64-le.
[   39.095264] systemd[1]: Running in initial RAM disk.
[   39.095454] systemd[1]: Set hostname to <ltc-test-ci2.aus.stglabs.ibm.com>.
[   39.148724] systemd[1]: Cannot add dependency job for unit blk-availability.service, ignoring: Unit not found.
[   39.149958] systemd[1]: Created slice Root Slice.
[   39.150027] systemd[1]: Starting Root Slice.
[   39.150187] systemd[1]: Listening on Journal Socket.
[   39.150249] systemd[1]: Starting Journal Socket.
[   39.150341] systemd[1]: Reached target Timers.
[   39.417008] tg3.c:v3.137 (May 11, 2014)
[   39.417068] pci 0005:02:09.0: enabling device (0141 -> 0143)
[   39.417144] tg3 0005:05:00.0: enabling device (0140 -> 0142)
[   39.419020] synth uevent: /devices/vio: failed to send uevent
[   39.419026] vio vio: uevent: failed to send synthetic uevent
[  OK  ] Started Device-Mapper Multipath Device Controller.
[  OK  ] Started Show Plymouth Boot Screen.
[  OK  ] Reached target Paths.
[  OK  ] Reached target Basic System.
[   39.449997] tg3 0005:05:00.0: Using 64-bit DMA iommu bypass
[   39.450672] tg3 0005:05:00.0 eth0: Tigon3 [partno(00RX892) rev 5719001] (PCI Express) MAC address 98:be:94:02:8f:64
[   39.450780] tg3 0005:05:00.0 eth0: attached PHY is 5719C (10/100/1000Base-T Ethernet) (WireSpeed[1], EEE[1])
[   39.450885] tg3 0005:05:00.0 eth0: RXcsums[1] LinkChgREG[0] MIirq[0] ASF[0] TSOcap[1]
[   39.450965] tg3 0005:05:00.0 eth0: dma_rwctrl[00000000] dma_mask[64-bit]
[   39.451214] tg3 0005:05:00.1: enabling device (0140 -> 0142)
[   39.455812] device-mapper: multipath service-time: version 0.3.0 loaded
[   39.491062] tg3 0005:05:00.1: Using 64-bit DMA iommu bypass
[   39.491705] tg3 0005:05:00.1 eth1: Tigon3 [partno(00RX892) rev 5719001] (PCI Express) MAC address 98:be:94:02:8f:65
[   39.491825] tg3 0005:05:00.1 eth1: attached PHY is 5719C (10/100/1000Base-T Ethernet) (WireSpeed[1], EEE[1])
[   39.491930] tg3 0005:05:00.1 eth1: RXcsums[1] LinkChgREG[0] MIirq[0] ASF[0] TSOcap[1]
[   39.492010] tg3 0005:05:00.1 eth1: dma_rwctrl[00000000] dma_mask[64-bit]
[   39.492252] tg3 0005:05:00.2: enabling device (0140 -> 0142)
[   39.501713] WARNING: CPU: 37 PID: 1241 at kernel/workqueue.c:1513 __queue_delayed_work+0xc8/0xf0
[   39.501807] Modules linked in: dm_service_time dm_multipath tg3(+)
[   39.501882] CPU: 37 PID: 1241 Comm: systemd-udevd Not tainted 4.16.0-1.rc7.dev.git58079f0.el7.centos.ppc64le #1
[   39.501978] NIP:  c00000000012d278 LR: c00000000012d2fc CTR: c00000000012d2a0
[   39.502049] REGS: c0000007ee793870 TRAP: 0700   Not tainted  (4.16.0-1.rc7.dev.git58079f0.el7.centos.ppc64le)
[   39.502143] MSR:  9000000000029033 <SF,HV,EE,ME,IR,DR,RI,LE>  CR: 8a002884  XER: 00000000
[   39.502220] CFAR: c00000000012d1e4 SOFTE: 1 
[   39.502220] GPR00: c00000000012d2fc c0000007ee793af0 c00000000146a600 c0000007f5690710 
[   39.502220] GPR04: c000000002fd0400 c0000007f56906f0 0000000000000000 0000000000000001 
[   39.502220] GPR08: 0000000000000000 c00000000012d170 0000000000000400 d00000000b9750d8 
[   39.502220] GPR12: c00000000012d2a0 c00000000fd59700 0000000100093ec0 0000000000000000 
[   39.502220] GPR16: 0000000100091300 0000000100091380 00000001000d0900 0000000100092f10 
[   39.502220] GPR20: 00000001000d0030 000001000e824667 000001000e8247a7 0000000000000007 
[   39.502220] GPR24: 0000000000000000 0000000000000000 c0000007ee793ca8 c0000007ee793ca0 
[   39.502220] GPR28: 0000000000000000 c0000007f15a0170 0000000000000000 0000000000000001 
[   39.502826] NIP [c00000000012d278] __queue_delayed_work+0xc8/0xf0
[   39.502886] LR [c00000000012d2fc] queue_delayed_work_on+0x5c/0x90
[   39.502946] Call Trace:
[   39.502971] [c0000007ee793af0] [c00000000012d2fc] queue_delayed_work_on+0x5c/0x90 (unreliable)
[   39.503057] [c0000007ee793b20] [d00000000b970cb8] __pg_init_all_paths+0x108/0x190 [dm_multipath]
[   39.503141] [c0000007ee793b60] [d00000000b970d8c] pg_init_all_paths+0x4c/0x80 [dm_multipath]
[   39.503225] [c0000007ee793ba0] [d00000000b972ac8] multipath_prepare_ioctl+0x138/0x150 [dm_multipath]
[   39.503310] [c0000007ee793bf0] [c0000000008e6900] dm_get_bdev_for_ioctl+0x120/0x1b0
[   39.503382] [c0000007ee793c40] [c0000000008e6d80] dm_blk_ioctl+0x50/0x110
[   39.503443] [c0000007ee793cc0] [c000000000595794] blkdev_ioctl+0x5f4/0xb80
[   39.503505] [c0000007ee793d20] [c0000000003df5c4] block_ioctl+0x54/0xa0
[   39.503566] [c0000007ee793d40] [c0000000003a02a4] do_vfs_ioctl+0xd4/0x8c0
[   39.503626] [c0000007ee793de0] [c0000000003a0b64] SyS_ioctl+0xd4/0x130
[   39.503688] [c0000007ee793e30] [c00000000000b8e0] system_call+0x58/0x6c
[   39.503748] Instruction dump:
[   39.503785] e8010010 7c0803a6 4e800020 60000000 60000000 60420000 7d435378 4bfff8c4 
[   39.503858] 0fe00000 4bffff98 0fe00000 4bffff80 <0fe00000> 4bffff6c 0fe00000 4bffff50 
[   39.503933] ---[ end trace 47786a0f55475f74 ]---
[   39.503985] WARNING: CPU: 37 PID: 1241 at kernel/workqueue.c:1515 __queue_delayed_work+0xb8/0xf0
[   39.504067] Modules linked in: dm_service_time dm_multipath tg3(+)
[   39.504130] CPU: 37 PID: 1241 Comm: systemd-udevd Tainted: G        W        4.16.0-1.rc7.dev.git58079f0.el7.centos.ppc64le #1
[   39.504235] NIP:  c00000000012d268 LR: c00000000012d2fc CTR: c00000000012d2a0
[   39.504307] REGS: c0000007ee793870 TRAP: 0700   Tainted: G        W         (4.16.0-1.rc7.dev.git58079f0.el7.centos.ppc64le)
[   39.504412] MSR:  9000000000029033 <SF,HV,EE,ME,IR,DR,RI,LE>  CR: 8a002884  XER: 00000000
[   39.504487] CFAR: c00000000012d200 SOFTE: 1 
[   39.504487] GPR00: c00000000012d2fc c0000007ee793af0 c00000000146a600 c0000007f5690710 
[   39.504487] GPR04: c000000002fd0400 c0000007f56906f0 0000000000000000 0000000000000001 
[   39.504487] GPR08: 0000000000000000 c0000007f56906f8 0000000000000400 d00000000b9750d8 
[   39.504487] GPR12: c00000000012d2a0 c00000000fd59700 0000000100093ec0 0000000000000000 
[   39.504487] GPR16: 0000000100091300 0000000100091380 00000001000d0900 0000000100092f10 
[   39.504487] GPR20: 00000001000d0030 000001000e824667 000001000e8247a7 0000000000000007 
[   39.504487] GPR24: 0000000000000000 0000000000000000 c0000007ee793ca8 c0000007ee793ca0 
[   39.504487] GPR28: 0000000000000000 c0000007f15a0170 0000000000000000 0000000000000001 
[   39.505091] NIP [c00000000012d268] __queue_delayed_work+0xb8/0xf0
[   39.505150] LR [c00000000012d2fc] queue_delayed_work_on+0x5c/0x90
[   39.505209] Call Trace:
[   39.505235] [c0000007ee793af0] [c00000000012d2fc] queue_delayed_work_on+0x5c/0x90 (unreliable)
[   39.505320] [c0000007ee793b20] [d00000000b970cb8] __pg_init_all_paths+0x108/0x190 [dm_multipath]
[   39.505403] [c0000007ee793b60] [d00000000b970d8c] pg_init_all_paths+0x4c/0x80 [dm_multipath]
[   39.505488] [c0000007ee793ba0] [d00000000b972ac8] multipath_prepare_ioctl+0x138/0x150 [dm_multipath]
[   39.505571] [c0000007ee793bf0] [c0000000008e6900] dm_get_bdev_for_ioctl+0x120/0x1b0
[   39.505644] [c0000007ee793c40] [c0000000008e6d80] dm_blk_ioctl+0x50/0x110
[   39.505705] [c0000007ee793cc0] [c000000000595794] blkdev_ioctl+0x5f4/0xb80
[   39.505766] [c0000007ee793d20] [c0000000003df5c4] block_ioctl+0x54/0xa0
[   39.505826] [c0000007ee793d40] [c0000000003a02a4] do_vfs_ioctl+0xd4/0x8c0
[   39.505887] [c0000007ee793de0] [c0000000003a0b64] SyS_ioctl+0xd4/0x130
[   39.505947] [c0000007ee793e30] [c00000000000b8e0] system_call+0x58/0x6c
[   39.506007] Instruction dump:
[   39.506043] 40de0050 4807fbed 60000000 38210020 e8010010 7c0803a6 4e800020 60000000 
[   39.506117] 60000000 60420000 7d435378 4bfff8c4 <0fe00000> 4bffff98 0fe00000 4bffff80 
[   39.506191] ---[ end trace 47786a0f55475f75 ]---
[   39.506242] WARNING: CPU: 37 PID: 1241 at kernel/workqueue.c:1444 __queue_work+0x160/0x5c0
[   39.506313] Modules linked in: dm_service_time dm_multipath tg3(+)
[   39.506375] CPU: 37 PID: 1241 Comm: systemd-udevd Tainted: G        W        4.16.0-1.rc7.dev.git58079f0.el7.centos.ppc64le #1
[   39.506481] NIP:  c00000000012cc80 LR: c00000000012cc54 CTR: c00000000012d2a0
[   39.506552] REGS: c0000007ee793790 TRAP: 0700   Tainted: G        W         (4.16.0-1.rc7.dev.git58079f0.el7.centos.ppc64le)
[   39.506657] MSR:  9000000000029033 <SF,HV,EE,ME,IR,DR,RI,LE>  CR: 2a002844  XER: 00000000
[   39.506732] CFAR: c000000000b637b4 SOFTE: 1 
[   39.506732] GPR00: c00000000012cc54 c0000007ee793a10 c00000000146a600 c0000007fc210800 
[   39.506732] GPR04: c0000007f56906f0 0000000000000000 0000000000000000 c000000001499d70 
[   39.506732] GPR08: 0000000000000000 0000000000000001 0000000000000000 d00000000b9750d8 
[   39.506732] GPR12: c00000000012d2a0 c00000000fd59700 0000000100093ec0 0000000000000000 
[   39.506732] GPR16: 0000000100091300 0000000100091380 c000000000d97f30 0000000000000001 
[   39.506732] GPR20: 0000000000000000 c000000000fbc8a8 c000000001624bd8 0000000000000000 
[   39.506732] GPR24: c000000001624bd0 c0000007ff526e00 0000000000000025 c000000000fbc8a8 
[   39.506732] GPR28: 0000000000000400 c000000002fd0400 c0000007f56906f0 c0000007e75b0000 
[   39.507336] NIP [c00000000012cc80] __queue_work+0x160/0x5c0
[   39.507384] LR [c00000000012cc54] __queue_work+0x134/0x5c0
[   39.507432] Call Trace:
[   39.507457] [c0000007ee793a10] [c00000000012cc54] __queue_work+0x134/0x5c0 (unreliable)
[   39.507530] [c0000007ee793af0] [c00000000012d2fc] queue_delayed_work_on+0x5c/0x90
[   39.507603] [c0000007ee793b20] [d00000000b970cb8] __pg_init_all_paths+0x108/0x190 [dm_multipath]
[   39.507687] [c0000007ee793b60] [d00000000b970d8c] pg_init_all_paths+0x4c/0x80 [dm_multipath]
[   39.507771] [c0000007ee793ba0] [d00000000b972ac8] multipath_prepare_ioctl+0x138/0x150 [dm_multipath]
[   39.507855] [c0000007ee793bf0] [c0000000008e6900] dm_get_bdev_for_ioctl+0x120/0x1b0
[   39.507927] [c0000007ee793c40] [c0000000008e6d80] dm_blk_ioctl+0x50/0x110
[   39.507988] [c0000007ee793cc0] [c000000000595794] blkdev_ioctl+0x5f4/0xb80
[   39.508049] [c0000007ee793d20] [c0000000003df5c4] block_ioctl+0x54/0xa0
[   39.508109] [c0000007ee793d40] [c0000000003a02a4] do_vfs_ioctl+0xd4/0x8c0
[   39.508170] [c0000007ee793de0] [c0000000003a0b64] SyS_ioctl+0xd4/0x130
[   39.508231] [c0000007ee793e30] [c00000000000b8e0] system_call+0x58/0x6c
[   39.508290] Instruction dump:
[   39.508326] 48a36b09 60000000 813f0018 2f890000 41de0314 60000000 7fc9f378 e9490009 
[   39.508400] 7d295278 7d290074 7929d182 69290001 <0b090000> 2fa90000 40de0360 815f0010 
[   39.508474] ---[ end trace 47786a0f55475f76 ]---
[   39.530965] tg3 0005:05:00.2: Using 64-bit DMA iommu bypass
[   39.531375] tg3 0005:05:00.2 eth2: Tigon3 [partno(00RX892) rev 5719001] (PCI Express) MAC address 98:be:94:02:8f:66
[   39.531479] tg3 0005:05:00.2 eth2: attached PHY is 5719C (10/100/1000Base-T Ethernet) (WireSpeed[1], EEE[1])
[   39.531581] tg3 0005:05:00.2 eth2: RXcsums[1] LinkChgREG[0] MIirq[0] ASF[0] TSOcap[1]
[   39.531657] tg3 0005:05:00.2 eth2: dma_rwctrl[00000000] dma_mask[64-bit]
[   39.531900] tg3 0005:05:00.3: enabling device (0140 -> 0142)
[   39.570962] tg3 0005:05:00.3: Using 64-bit DMA iommu bypass
[   39.571363] tg3 0005:05:00.3 eth3: Tigon3 [partno(00RX892) rev 5719001] (PCI Express) MAC address 98:be:94:02:8f:67
[   39.571469] tg3 0005:05:00.3 eth3: attached PHY is 5719C (10/100/1000Base-T Ethernet) (WireSpeed[1], EEE[1])
[   39.571572] tg3 0005:05:00.3 eth3: RXcsums[1] LinkChgREG[0] MIirq[0] ASF[0] TSOcap[1]
[   39.571649] tg3 0005:05:00.3 eth3: dma_rwctrl[00000000] dma_mask[64-bit]
[   39.573648] tg3 0005:05:00.0 enP5p5s0f0: renamed from eth0
[   39.762117] tg3 0005:05:00.3 enP5p5s0f3: renamed from eth3
[   39.822009] tg3 0005:05:00.1 enP5p5s0f1: renamed from eth1
[   39.911987] tg3 0005:05:00.2 enP5p5s0f2: renamed from eth2
````

**Comment by cdeadmin** [comment]
------- Comment From bssrikanth@in.ibm.com 2018-04-02 00:40:01 EDT-------
Are there are one looking into this?

**Comment by cdeadmin** [comment]
------- Comment From pmac@au1.ibm.com 2018-04-02 19:18:29 EDT-------
This is an upstream bug that got fixed just in time for the 4.16 release. I'll merge in 4.16 today and push it out, and that should fix this bug.

------- Comment From seg@us.ibm.com 2018-04-02 19:20:48 EDT-------
Moving to fixedawaitingtest state, as this should appear in tomorrow's daily build.

**Comment by cdeadmin** [comment]
------- Comment From satheera@in.ibm.com 2018-04-09 01:27:55 EDT-------
Tested booting fine,

# uname -r
4.16.0-2.dev.gitb24758c.el7.centos.ppc64le

# lscpu
Architecture:          ppc64le
Byte Order:            Little Endian
CPU(s):                160
On-line CPU(s) list:   0,8,16,24,32,40,48,56,64,72,80,88,96,104,112,120,128,136,144,152
Off-line CPU(s) list:  1-7,9-15,17-23,25-31,33-39,41-47,49-55,57-63,65-71,73-79,81-87,89-95,97-103,105-111,113-119,121-127,129-135,137-143,145-151,153-159
Thread(s) per core:    1
Core(s) per socket:    5
Socket(s):             4
NUMA node(s):          4
Model:                 2.1 (pvr 004b 0201)
Model name:            POWER8E (raw), altivec supported
CPU max MHz:           3690.0000
CPU min MHz:           2061.0000
Hypervisor vendor:     (null)
Virtualization type:   full
L1d cache:             64K
L1i cache:             32K
L2 cache:              512K
L3 cache:              8192K
NUMA node0 CPU(s):     0,8,16,24,32
NUMA node1 CPU(s):     40,48,56,64,72
NUMA node16 CPU(s):    80,88,96,104,112
NUMA node17 CPU(s):    120,128,136,144,152

Regards,
-Satheesh
